R/calc_srtmElevation.R

```diff
@@ -0,0 +1,86 @@
+#' Calculate elevation based on SRTM rasters
+#'
+
+.calc_SRTMElevation <- function(shp,
+                                srtmelevation,
+                                engine = "zonal",
+                                stats = "mean",
+                                rundir = tempdir(),
+                                verbose = TRUE,
+                                todisk = FALSE,
+                                ...) {
+
+  if (ncell(srtmelevation) > 1024 * 1024) todisk <- TRUE
+  if (engine == "extract") {
+    tib.zstats <- .comp_eleExtract(
+      elevation = srtmelevation,
+      shp = shp,
+      stats = stats
+    )
+    return(tib.zstats)
+
+  } else {
+    tib.zstats <- .comp_eleZonal(
+      elevation = srtmelevation,
+      shp = shp,
+      stats = stats,
+      todisk = todisk,
+      rundir = rundir
+    )
+    return(tib.zstats)
+  }
+}
+
+
+#' function to compute single raster with terra extract
+#'
+
+.comp_eleExtract <- function(elevation = NULL,
+                             shp = NULL,
+                             stats = "mean",
+                             ...) {
+  shp.v <- vect(shp)
+  zstats <- terra::extract(elevation,
+    shp.v,
+    fun = stats,
```

Comment:
I suggest to also allow users to compute mean/median/sd for the DEM. Similar to the comments in the TRI routine, this should include a check that the stats were correctly specified and only calculate the one that a user is interested in.